docs/arch/agent.md

```diff
@@ -0,0 +1,45 @@
+
+# Summary
+The agent is the powerhouse of SealCi. It will receive jobs and try to run them in order to complete the operational part of the CI.
+
+# References
+
+## Jobs
+Jobs are process that are launched in a controlled environment provided from a configuration. Those process are list of choosen commands.
+
+## Artifacts
+Artifacts is the result of the execution of a job. We consider as artifacts:
+- logs of execution of the pod
+- any file that is created during the execution
+
+
+# Agent operations
+
+## Life of an agent
+
+**Launching & registering to the scheduler**
+At its start the agent will try to register to the scheduler in order to accept new jobs. For the acknowledgement, the agent will init a connection that will allow both the agent the scheduler to stream datas to each other.   
+
+**Health and death**
+During the life of the agent, it will stream regularly health information of the agent. The scheduler will know that an agent is dead once the stream between them is dead.
```

Comment:
There is usually no regularity in how the agent updates the scheduler with it status. The agent must only do so when its status changes.

```suggestion
An agent streams health and status information to the scheduler, and the agent is kept on the scheduler's resource pool as long as it maintains its connection with it.
```

Reply:
I have a question here. We don't really get what is a status change ? 

For me the most practical thing is to inform the scheduler each time the cpu or ram change in a certain range. Like 15% to 20% then 20% to 25%. I'm not sure it is what you meant.

We could also inform the scheduler each time a step is launched our health information. But then some steps could take a lot of time to complete and therefore begin to consume a lot of ram and cpu at any moment. 
In this case the cpu or ram usage could be 100% but the last stream told the scheduler 25%, it could be critical.

That is why we decided to stream periodically our health information. We don't get how other solution could be better.

Reply:
A status change is an agent's state change. When the agent capacity changes in a meaningful way, or when the agent knows that it must allocate a certain capacity for handling a step or for anything else, it must notify the scheduler about it. For example, when the scheduler sends a step that consumes n cpus and M slices of memory, the agent takes that information into account and reserves that amount of resources + whatever overhead it needs to actually run the step. Then it informs the scheduler about that status change.
That's just one example, there could be other reasons for an agent capcity to drop or increase, but that's for the agent to manage.